lib/logstash/inputs/elasticsearch.rb

```diff
@@ -350,24 +350,42 @@ def run(output_queue)
   end
 
   ##
+  # @param output_queue [#<<]
+  # @param scroll_id [String]: a scroll id to resume
+  # @return [Array(Boolean,String)]: a tuple representing whether the response
+  #
+  def process_next_scroll(output_queue, scroll_id)
+    r = scroll_request(scroll_id)
+    r['hits']['hits'].each { |hit| push_hit(hit, output_queue) }
+    [r['hits']['hits'].any?, r['_scroll_id']]
+  end
+
   # This can be called externally from the query_executor
   public
   def push_hit(hit, output_queue, root_field = '_source')
-    event = targeted_event_factory.new_event hit[root_field]
-    set_docinfo_fields(hit, event) if @docinfo
+    event = event_from_hit(hit, root_field)
     decorate(event)
     output_queue << event
   end
 
+  def event_from_hit(hit, root_field)
+    event = targeted_event_factory.new_event hit[root_field]
+    set_docinfo_fields(hit, event) if @docinfo
+
+    event
+  rescue => e
```

Comment:
```suggestion
  rescue RuntimeError => e
```
Looks like `set_docinfo_fields` explicitly raises a `RuntimeError` now? May be best to be as specific as possible with error catching.

Reply:
Yes, but this isn't the only exception we're handling here.

We know that `EventFactory#new_event` (which delegates to `Event::new`) can give us exceptions too, specifically when the payload uses reserved fields in incorrect ways. We want to catch any situation in which we fail to create an event from the hit.

Reply:
I see, Ok yeah that makes sense.